java/src/jmri/implementation/AbstractTurnout.java

```diff
@@ -111,6 +112,7 @@ public int getKnownState() {
     public void setCommandedState(int s) {
         log.debug("set commanded state for turnout {} to {}", getFullyFormattedDisplayName(),
                 (s==Turnout.CLOSED ? closedText : thrownText));
+        waitOutputInterval(); // if interval > 0, wait before next output command
```

Comment:
I don't think this does what you want it to.

What this says is when we setCommandedState, we will ALWAYS wait as specified in waitOutputInterval before forwarding the command to the layout regardless of what the interval is between calls to setCommandedState.  What we really want is to wait only when the interval between calls to setCommandedState is less than the wait time specified in waitOutputInterval.

In fact, I don't think this even ensures that there will be a minimum spacing between commands issued from multiple threads.  Let's say we have two different scripts executing on their own threads (scripts do that).  If both call setCommandedState (possibly for different turnouts) at nearly the same time, both threads will wail wait the specified interval and then continue at basically the same time to forward the state change on to the layout.

Reply:
From what @balazsracz says above, I learn that ``setCommandedState()`` is "by the threading model of JMRI always supposed to be called on the layout thread, which at present moment is the AWT event thread" and that ``sleep()`` would "hang" it (but isn't it still on its own thread, and can't it be interrupted?).
Now @pabender describes the behavior like I envisaged it as "both threads will wait the specified interval (...) and then continue (...) to forward the state change onto the layout" and that is diffent.
My simple suggestion does not do stuff like manage some sort of AWT event _Buffer_ with its own internal timer to release each to the layout. Instead, I just want the known sources of intense series of output comands such as the MatrixSignalMast to slow down their for-loop. We might use ``t.getOutputInterval()`` in other places as well.

Reply:
Let me try to explain the biggest issue I have with this...

You click the button to throw a turnout.... then several seconds later click a button to throw a turnout ( maybe the same one, but maybe a different one ). Both operations have to wait the specified interval, but you really don’t want either operation to wait because the time between operations is larger than the specified interval.

To do this right, the wait has to be conditional based on the time since the last operation.

For the multiple caller situation, the waitOutputInterval method needs to be synchronized, otherwise you have no way to hold of one caller while the other waits.

Reply:
This now works as it should by getting the earliest next moment to send from the TurnoutManager running on a given connection:
![turnoutoutputinterval](https://user-images.githubusercontent.com/14811327/49704064-57f1b480-fc0e-11e8-9cd5-3675e7c72494.png)

Reply:
@pabender can you approve? Don't want to use ``Dismiss review``. The method of timing has been changed as suggested since commit https://github.com/JMRI/JMRI/pull/6177/commits/1845a70e09d1358650b8e8c3116a218f559e0d66